circleci/resource_circleci_environment_variable.go

```diff
@@ -161,21 +181,23 @@ func resourceCircleCIEnvironmentVariableCreate(d *schema.ResourceData, m interfa
 	envName := d.Get("name").(string)
 	envValue := d.Get("value").(string)
 
-	exists, err := providerClient.EnvVarExists(organization, projectName, envName)
-	if err != nil {
-		return err
-	}
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		exists, err := providerClient.EnvVarExists(organization, projectName, envName)
+		if err != nil {
+			return wrap(err)
+		}
 
-	if exists {
-		return fmt.Errorf("environment variable '%s' already exists for project '%s'", envName, projectName)
-	}
+		if exists {
+			return wrap(fmt.Errorf("environment variable '%s' already exists for project '%s'", envName, projectName))
+		}
 
-	if _, err := providerClient.AddEnvVar(organization, projectName, envName, envValue); err != nil {
-		return err
-	}
+		if _, err := providerClient.AddEnvVar(organization, projectName, envName, envValue); err != nil {
+			return wrap(err)
+		}
```

Comment:
Should we split the retry here and use a different, one for the call to `EnvVarExists()` and another retry for `AddEnvVar()` ? 

If the first one succeed (and return false) but the second one fail due to network issue (aka a retryable error), should we do the first request again too?
I'm not sure if it's overkill to do 2 different retries and what we gain from doing only one.

What do you think?

Reply:
splitting makes sense to me